packages/amplify-cli-core/src/utils/packageManager.ts

```diff
@@ -1,39 +1,115 @@
 import * as fs from 'fs-extra';
 import * as path from 'path';
-import * as which from 'which';
+import which from 'which';
 import { coerce, SemVer } from 'semver';
 import { execWithOutputAsString } from './shell-utils';
+import { AmplifyError } from '../errors/amplify-error';
+import { BuildType } from '@aws-amplify/amplify-function-plugin-interface';
 
 /**
  * package managers type
  */
-export type PackageManagerType = 'yarn' | 'npm';
-
+export type PackageManagerType = 'yarn' | 'npm' | 'pnpm' | 'custom';
```

Comment:
Isn't this package manager also used for compiling overrides/custom resources? Will this PR enable compiling the overrides/custom as well?

Reply:
cc: @akshbhu

Reply:
this seems to be the case, I was not aware of.

https://github.com/lazpavel/amplify-cli/blob/lambda-package-manager/packages/amplify-category-custom/src/utils/build-custom-resources.ts#L71-L94

Reply:
This seems like it can break on cli if Cx chooses ```custom``` type  for a function.

Reply:
if the Cx chooses custom for a function, only that function will be built with `custom` not other functions or custom resources, for custom resources the prompt is not presented and the decision will be made based on the custom resource directory config, which lock files exist, if non exist, it will pick first available executable in the system: yarn, pnpm, npm in this order of testing, also npm is used regardless if it exists or not in the system (old behavior)